qa/os/src/test/java/org/elasticsearch/packaging/test/PackagingTestCase.java

```diff
@@ -392,4 +394,32 @@ public static Path createTempDir(String prefix) throws IOException {
         return Files.createTempDirectory(getRootTempDir(), prefix, NEW_DIR_PERMS);
     }
 
+    @FunctionalInterface
+    public interface ThrowingConsumer<T> {
```

Comment:
We probabaly don't want to move this interface, since it's only being used in this file, but a possible alternative home would be `libs/core/src/main/java/org/elasticsearch/common`, since we already have `CheckedFunction` and `CheckedRunnable` in there.

Reply:
We had CheckedConsumer in server, but I've now moved it to elasticsearch-core lib.